hub/core/storage/provider.py

```diff
@@ -60,6 +93,23 @@ def __setitem__(
         Raises:
             InvalidBytesRequestedError: If `start_byte` < 0.
         """
+        start_byte = start_byte or 0
```

Comment:
I like this way of doing things, but we should document it better. Right now the docstring lines for `set_bytes` and `__setitem__` are identical